crates/driver/src/tests/setup/mod.rs

```diff
@@ -273,6 +274,15 @@ impl Order {
         }
     }
 
+    pub fn pre_interaction(self, interaction: Interaction) -> Self {
+        let mut pre_interactions = self.pre_interactions;
+        pre_interactions.push(interaction);
```

Comment:
nit: Could be simpler without destructuring
```suggestion
    pub fn pre_interaction(mut self, interaction: Interaction) -> Self {
        self.pre_interactions.push(interaction);
        self
```